### PR TITLE
feat: docker compose integration [pt. 1]

### PR DESCRIPTION
**Pull request by tedim52**
## Description:
This PR is the pt. 1 of enabling the use case:
```
kurtosis run docker-compose.yaml
```
This PR contains only the logic for transpiling docker compose into starlark. The next PR will integrate it with APIC.

## Is this change user facing?
NO (next one will be).

### CODE REVIEW

File: core/server/api_container/server/docker_compose_transpiler/docker_compose_transpiler.go
Comment (lostbean) [inline core/server/api_container/server/docker_compose_transpiler/docker_compose_transpiler.go]: do we really need `envVars` here? Why not just use `envVarsInFile`

File: core/server/api_container/server/docker_compose_transpiler/docker_compose_transpiler.go
Comment (lostbean) [inline core/server/api_container/server/docker_compose_transpiler/docker_compose_transpiler.go]: Wondering if this kind of configuration should go in the `const` section at the top.

File: core/server/api_container/server/docker_compose_transpiler/docker_compose_transpiler.go
Comment (lostbean) [inline core/server/api_container/server/docker_compose_transpiler/docker_compose_transpiler.go:434]: These kind of encoding seems a bit trick and can potentially lead to some confusion with user. Wondering if we should simplify the conversion to a safer transcription and let the user adjust the richer options in Starlark later.

File: core/server/api_container/server/docker_compose_transpiler/docker_compose_transpiler.go
Comment (lostbean) [inline core/server/api_container/server/docker_compose_transpiler/docker_compose_transpiler.go:538]: Nice :)

**Review by lostbean — APPROVED** [review]
Left some small comments but looks good

File: core/server/api_container/server/docker_compose_transpiler/docker_compose_transpiler.go
Comment (tedim52) [reply core/server/api_container/server/docker_compose_transpiler/docker_compose_transpiler.go:434]: Definitely. Particularly with volume's, it's not a 1:1 match in terms of semantics, but rather it requires translating Compose's volume concept into our concepts which definitely needs to be transparent to users. 

Going to put some thought into this, but do you have any ideas for what a simpler transcription might be?

File: core/server/api_container/server/docker_compose_transpiler/docker_compose_transpiler.go
Comment (tedim52) [reply core/server/api_container/server/docker_compose_transpiler/docker_compose_transpiler.go]: facts

File: core/server/api_container/server/docker_compose_transpiler/docker_compose_transpiler.go
Comment (tedim52) [reply core/server/api_container/server/docker_compose_transpiler/docker_compose_transpiler.go]: facts

File: core/server/api_container/server/docker_compose_transpiler/docker_compose_transpiler.go
Comment (tedim52) [reply core/server/api_container/server/docker_compose_transpiler/docker_compose_transpiler.go:434]: After putting some thought into it, going to merge this for now and return to this when writing docs on the translation between files artifact and volume.

**Pull request merged**
